pkg/service/sniffer/privileged_pod_sniffer_service.go

```diff
@@ -27,21 +28,42 @@ func (p *PrivilegedPodSnifferService) Setup() error {
 
 	log.Infof("creating privileged pod on node: '%s'", p.settings.DetectedPodNodeName)
 
-	p.privilegedPod, err = p.kubernetesApiService.CreatePrivilegedPod(p.settings.DetectedPodNodeName, p.settings.Image)
+	bridge := runtime.NewContainerRuntimeBridge(p.settings.DetectedContainerRuntime)
+	p.runtimeBridge = bridge
```

Comment:
I tend to believe this code should be part of this struct constructor, or even better - we should have the runtime bridge passed as a parameter to this struct constructor (DI)

Reply:
The runtime bridge has a dependency on the detected runtime which is pulled from the pod at the end of `Validate`. I'm not quite sure how to rearrange that but still looking.

Reply:
This is now resolved